base/atomics.jl

```diff
@@ -202,6 +202,22 @@ inttype(::Type{Float16}) = Int16
 inttype(::Type{Float32}) = Int32
 inttype(::Type{Float64}) = Int64
 
+function alignment{T}(::Type{T})
+    # Atomic types assume the same alignment with the corresponding non-atomic
+    # types, with the following exceptions: On 32- and 64-bit x86 platforms
+    # and on 64-bit SPARC platforms, atomic types of size 1, 2, 4, 8 or 16 -byte
+    # have the alignment that matches the size.
+    #
+    # Note for PPC:
+    # The OpenPower ABI for Linux Supplement -- 64-Bit ELF V2 ABI Specification
+    # defines that the alignments of types is sizeof(T).
+    if Sys.ARCH === :x86_64 || Sys.ARCH === :i686 ||
```

Comment:
The alignment here should not be what the ABI require, but what we can guarantee. See `jl_gc_alignment` for the correct definition.